Resolved all the issues

KTSLX opened the pull request:
Работа над ошибками:

Переработал файл locators.py - все локаторы имеют свою переменную, убрал неиспользуемые локаторы.
Изменил все локаторы (отошёл от использования порядковых номеров DOM, перестал использовать абсолютные пути, привязался к конкретным атрибутам элементов)
Модуль login_module.py оставил как есть (не стал переносить в фикстуры), т.к. он используется не во всех тестах.
Убрал из фикстуры driver ненужное ожидание появления элемента
Добавил ассерты во всех тестах где не было.
Тест навигации в конструкторе разбил на 3 отдельных используя параметризацию.
Все тесты используют локаторы из переменных в файле locators.py
Убрал принты из всех тестов
Добавил модуль data_module.py в котором хранятся данные для логина/регистрации

KTSLX committed:
Added tests and README.MD

KTSLX committed:
Переработал файл locators.py - все локаторы имеют свою переменную, убрал неиспользуемые локаторы.
Изменил все локаторы (отошёл от использования порядковых номеров DOM, перестал использовать абсолютные пути, привязался к конкретным атрибутам элементов)
Модуль login_module.py оставил как есть (не стал переносить в фикстуры), т.к. он используется не во всех тестах.
Убрал из фикстуры driver ненужное ожидание появления элемента
Добавил ассерты во всех тестах где не было.
Тест навигации в конструкторе разбил на 3 отдельных используя параметризацию.
Все тесты используют локаторы из переменных в файле locators.py
Убрал принты из всех тестов
Добавил модуль data_module.py в котором хранятся данные для логина/регистрации

AlexKitIt commented on locators.py:
```diff
@@ -0,0 +1,39 @@
+
+#Логин
+email_input_field = './/main/div/form/fieldset/div/div/input[@type="text"]'  #поле ввода email на странице входа
```
Нужно исправить: здесь и далее: использование порядкового номера элемента в dom является весьма нестабильным способом поиска, необходимо привязываться к атрибутам конкретного элемента
Кроме того ходить от main и использовать абсолютные пути нельзя - очень ненадежно, добавится на пути новый элемент - тест ляжет
все подобные локаторы необходимо переписать, кроме того, лучше их хранить в кортеже с методом поиска, так будет удобнее поддерживать

KTSLX replied:
Исправил - максимально сократил все XPATHы.

AlexKitIt commented on login_module.py:
```diff
@@ -0,0 +1,16 @@
+from selenium.webdriver.support.ui import WebDriverWait
+import locators
+import data_module
+def login(driver):
```
Можно лучше: очень похоже на фикстуру

KTSLX replied:
Применяется не во всех тестах, по-этому и отказался от переноса в фикстуры.
Или сделать условие в фикстуре по названию теста?

AlexKitIt replied:
Условий в фикстурах быть не может, порядок вызова определяется в тестах, если фикстура нужна - она туда передается, так что разницы нет, в 1-2 тестах она используется или во всех. В этом проекте можно оставить, далее - нельзя

KTSLX replied:
Понял, спасибо

AlexKitIt commented on tests/test_constructor_navigation.py:
```diff
@@ -0,0 +1,34 @@
+        (locators.buns_button, locators.buns_header, locators.constructor_window)  # Булки (со скроллом)
+    ]
+)
+def test_constructor_navigation(driver, button_xpath, header_xpath, scroll_xpath):
```
Нужно исправить: здесь и далее по тестам: тестовые методы в модулях объединяются в тестовые классы

KTSLX replied:
Не понимаю что нужно переделать

KTSLX replied:
О каких методах идёт речь?

KTSLX replied:
(locators.buns_button, locators.buns_header, locators.constructor_window) - это не методы, а переменные с локаторами

KTSLX replied:
Кажется я понял о чём речь.
Я немного отстаю и только сейчас прохожу page object model

KTSLX replied:
Если что, это проект 5 спринта

AlexKitIt replied:
Коммент править обязательно. Речь идет о том, что в каждом тестом модуле создается класс по синтаксису pytest по форме Test.. а в него уже размещаются тестовые методы test_... К номеру проекта это отношения не имеет

KTSLX replied:
Исправил - разбил на три теста и сделал их наследниеками тествого класса

AlexKitIt commented on tests/test_constructor_navigation.py:
```diff
@@ -0,0 +1,34 @@
+def test_constructor_navigation(driver, button_xpath, header_xpath, scroll_xpath):
+
+    # Описываем условие для скролла (нужен для теста булок, т.к. булки видны изначально)
+    if scroll_xpath:
```
Нужно исправить: здесь и далее: условий в тестах быть не может, тест всегда однозначен

KTSLX replied:
Предлагаете сделать прокрутку для каждого теста, а не только для одного?

KTSLX replied:
scroll_xpath это один из параметров в параметризации. Нужен чтобы сделать прокрутку  только для одного теста(для булок, т.к. конструктор уже отображает булки). Для двух тестов он задан как None, а для булок - локатор окна прокрутки
Не понимаю почему нельзя использовать такое условие.

AlexKitIt replied:
Коммент править обязательно. Тест - это не функциональное программирование, кроме того условие тут используется чтобы уместить тесты в параметризацию. Если они не влезают сюда - значит это несколько тестов.

KTSLX replied:
Разбил тест на три отдельных, чтобы избавиться от условия, к тому же тесты являются частью тестового класса TestConstructorNavigation:

AlexKitIt commented on tests/conftest.py:
```diff
@@ -0,0 +1,13 @@
+@pytest.fixture(scope="function")
+def driver():
+    driver = webdriver.Chrome()
+    driver.get("https://stellarburgers.nomoreparties.site/")
```
Можно лучше: урлы стоит хранить отдельно, они постоянно переиспользуются

KTSLX replied:
В этом проекте всего один урл

KTSLX replied:
Добавил файл config с урлом в переменной. Фикстуру тоже изменил, чтобы брала УРЛ из переменной

KTSLX committed:
В тесте test_navigate_to_burger_constructor.py убрал ненужную строку с ожиданием появления кнопки (она уже есть в модуле login.py
Переработал локаторы - сократил XPATHы насколько возможно

KTSLX committed:
Добавил файл config, в котором хранится переменная для фикстуры
Изменил фикстуру чтобы использовала URL из переменной.

KTSLX committed:
Ещё сократил XPATHы из раздела Логин

AlexKitIt commented on tests/test_login.py:
```diff
@@ -0,0 +1,42 @@
+    ]
+)
+def test_login_different_start_points_logged_in(driver, test_type, login_button, login_start_page):
+    if test_type == 'main_page':
```
Нужно исправить: эти условия так же выпиливать обязательно. У теста не может быть ветвлений ни хода выполнения ни проверки. Это несколько тестов. Параметризация это инструмент облегчающий жизнь, но использовать его нужно с умом и только там где уместно

KTSLX replied:
Убрал условия и разбил тест на 4 разных (без использования параметризации). 
Все тесты принадлежат тестовому классу TestLogin

KTSLX committed:
Переделал все тесты чтобы каждый тестовый метод был наследником тестового класса.
Переделал test_constructor_navigation.py, test_navigate_to_burger_constructor.py и test_login.py, чтобы в них не было условий (разбил на отдельные тесты).

KTSLX commented:
Спасибо за ревью, всё исправил
